obspy/io/sac/arrayio.py

```diff
@@ -0,0 +1,637 @@
+# -*- coding: utf-8 -*-
+"""
+Low-level array interface to the SAC file format.
+
+Functions in this module work directly with numpy arrays that mirror the SAC
+format.  The 'primitives' in this module are the float, int, and string header
+arrays, the float data array, and a header dictionary. Convenience functions
+are provided to convert between header arrays and more user-friendly
+dictionaries.
+
+These read/write routines are very literal; there is almost no value or type
+checking, except for byteorder and header/data array length.  File- and array-
+based checking routines are provided for additional checks where desired.
+
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.utils import native_str
+from future.builtins import *
+
+import sys
+import warnings
+
+import numpy as np
+
+from obspy.core.compatibility import from_buffer
+
+from ..sac import header as HD
+from .util import SacIOError, SacInvalidContentError
+from .util import is_valid_enum_int, is_same_byteorder, byteswap
+
+
+def init_header_arrays(*args):
+    """
+    Initialize arbitrary header arrays.
+
+    Parameters
+    ----------
+    which : str, {'float', 'int', 'str'}
+        Specify which arrays to initialize, and desired order.
+        If omitted, returned arrays are ('float', 'int', 'str').
+
+    Returns
+    -------
+    arrays : list of numpy.ndarrays
+        The desired SAC header arrays.
+
+    """
+    if len(args) == 0:
+        args = ('float', 'int', 'str')
+    #http://stackoverflow.com/a/13052254/745557
+    out = []
+    for itype in args:
+        if itype == 'float':
+            # null float header array
+            # XXX: why not init native byte order?
+            hf = np.ndarray(70, dtype=native_str('<f4'))
+            hf.fill(HD.FNULL)
```

Comment:
This is a bit obscure to read. Can you use `np.full()` which is only marginally slower but much easier to read in my opinion:

```python
In [2]: %timeit a = np.empty(1e6, dtype=np.float32); a.fill(5.0)
The slowest run took 4.91 times longer than the fastest. This could mean that an intermediate result is being cached
1000 loops, best of 3: 344 µs per loop

In [5]: %timeit a = np.full(1e6, fill_value=5.0, dtype=np.float32)
1000 loops, best of 3: 355 µs per loop
```